pkg/odo/genericclioptions/context.go

```diff
@@ -257,3 +260,63 @@ func ignoreButLog(err error) {
 		glog.V(4).Infof("Ignoring error as it usually means flag wasn't set: %v", err)
 	}
 }
+
+// LocalConfigInfo is a struct that will contain the LocalConfig as well as SourcePath
```

Comment:
Are you sure we should be inserting it into here:

```sh
// GetSourceLocation returns the sourcelocation, returns default if nil                                                               
func (lc *LocalConfig) GetSourceLocation() string {                                                                                   
  if lc.componentSettings.SourceLocation == nil {                                                                                     
    return ""                                                                                                                         
  }
  return *lc.componentSettings.SourceLocation
}
```

This would change the functionality with regards to how the other `Get()` functions work..if you're okay with that I'll go forward with pushing it to: `GetSourceLocation()`